yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala

```diff
@@ -53,4 +62,61 @@ private[spark] class YarnClusterSchedulerBackend(
       logError("Application attempt ID is not set.")
       super.applicationAttemptId
     }
+
+  override def getDriverLogUrls: Option[Map[String, String]] = {
+    var yarnClientOpt: Option[YarnClient] = None
+    var driverLogs: Option[Map[String, String]] = None
+    val appId: String = applicationId()
+    if (appId == null) {
```

Comment:
This will never be `null` (see `SchedulerBackend.scala`). If you want to do this check you'd have to explicitly check the value in `sc.getConf`.